polaris-core/src/main/java/org/apache/polaris/core/persistence/cache/EntityCache.java

```diff
@@ -53,16 +52,22 @@ public class EntityCache {
   // index by name
   private final AbstractMap<EntityCacheByNameKey, ResolvedPolarisEntity> byName;
 
+  private final PolarisCallContext polarisCallContext;
+
   /**
    * Constructor. Cache can be private or shared
    *
    * @param polarisMetaStoreManager the meta store manager implementation
    */
-  public EntityCache(@Nonnull PolarisMetaStoreManager polarisMetaStoreManager) {
+  public EntityCache(
+      @Nonnull PolarisMetaStoreManager polarisMetaStoreManager,
+      @Nonnull PolarisCallContext polarisCallContext) {
 
     // by name cache
     this.byName = new ConcurrentHashMap<>();
 
+    this.polarisCallContext = polarisCallContext;
```

Comment:
EntityCache can't have a PolarisCallContext in its constructor and certainly not as an instance field. The cache lives beyond a single call, so only long lived contexts should be referenced. Any number of things can change between calls, even things that might change the configuration store's response.

Reply:
cc @gh-yzou who I believe is working on an application-scoped feature store